### ISSUE TITLE
"_module" error

### ISSUE THREAD

**Issue by Knockback2003** [sdaoudi/HomeAssistant-GDrive-Uploader#9]
After changing the secret_file_path to the credentials_file_path and place my credentials.json from google api in their a error "failded to call script ... "_module" "

**Comment by sdaoudi** [comment]
How did you generated the `credentials.json` file?
Could you provide me more logs please?

For your information, you can follow this documentation to generate the credentials.json file:
https://github.com/sdaoudi/HomeAssistant-GDrive-Uploader/blob/main/SERVICE_ACCOUNT.md

**Comment by KierewietNL** [comment]
I have the same problem.

Here you can see the complete logging:
```
Logger: homeassistant.helpers.script.websocket_api_script
Bron: helpers/script.py:525
Eerst voorgekomen: 16:40:26 (3 gebeurtenissen)
Laatst gelogd: 16:58:14

websocket_api script: Error executing script. Unexpected error for call_service at pos 1: '_module'
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 525, in _async_step
    await getattr(self, handler)()
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 763, in _async_call_service_step
    response_data = await self._async_run_long_action(
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 726, in _async_run_long_action
    return await long_task
           ^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/core.py", line 2761, in async_call
    response_data = await coro
                    ^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/core.py", line 2811, in _execute_service
    return await self._hass.async_add_executor_job(target, service_call)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/gdrive_uploader/__init__.py", line 30, in handle_upload
    gdrive = GDriveApi(credentials_file_path)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/gdrive_uploader/api.py", line 14, in __init__
    gauth.LoadCredentialsFile(credentials_file_path)
  File "/usr/local/lib/python3.12/site-packages/pydrive2/auth.py", line 418, in LoadCredentialsFile
    self.credentials = self._default_storage.get()
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/oauth2client/client.py", line 407, in get
    return self.locked_get()
           ^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/oauth2client/file.py", line 54, in locked_get
    credentials = client.Credentials.new_from_json(content)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/oauth2client/client.py", line 302, in new_from_json
    module_name = data['_module']
                  ~~~~^^^^^^^^^^^
KeyError: '_module'
```

I generated the credentials.json as you described in https://github.com/sdaoudi/HomeAssistant-GDrive-Uploader/blob/main/SERVICE_ACCOUNT.md
Although your instructions did not exactly macth the GUI.

I also noticed some differnces in your instructions: "The credentials_file_path is the path to your credentials.json file generated with your client_id and client_secret"

My JSON file has no field named "client_secret".  I only see the following fields:

- type
- project_id
- private_key_id
- private_key
- client_email
- client_id
- auth_uri
- token_uri
- auth_provider_x509_cert_url
- client_x509_cert_url
- universe_domain

Did I generate the right JSON file?

**Comment by KierewietNL** [comment]
I tried some googling myself (but I am completely unknown to the subject so forgive me if I am talking nonsense :) ).

Apparently, the JSON I provided is not in the right format. The JSON I need needs to be in the "oauth" format. 
The problem is that I am not able to get a JSON file from the service account in the right format. 

I am able to create an OAuth 2.0 Client ID, but this ID doesn't has an e-mail adres included which is needed to give acces to the Google Drive folder. 

So currently I am stuck here.
 
![image](https://github.com/user-attachments/assets/49932dc4-a7da-4f95-9aa0-3ce109bf259c)

Am I doing something wrong?

**Comment by KierewietNL** [comment]
I found the problem. I installed the integration via de the custom repo. Apparently this is a different version then the one that is found under: https://github.com/sdaoudi/HomeAssistant-GDrive-Uploader/releases/latest

So I completely removed the integration and did a manual installation. Besides that I had to change the path in configuration.yaml from:

```
gdrive_uploader:
  credentials_file_path: "/config/secrets/credentials.json"
```
to 
```
gdrive_uploader:
  secret_file_path: "/config/secrets/credentials.json"
```
and now it works like a charm

**Comment by sdaoudi** [comment]
Good news  👍 

I see that google changed how the keys are created for service account. 

I updated documentation in this PR #10

**Comment by sdaoudi** [comment]
I published a new version [v1.1.0](https://github.com/sdaoudi/HomeAssistant-GDrive-Uploader/releases/tag/v1.1.0).

- The 2 credentials_file_path and secret_file_path configurations are supported
- The 2 access token and service account formats are supported

This new version should fix all problems. I'll let you create a new ticket if you have any other issues.

Thanks
